### PR TITLE
chore(core): typehinting more various lib functions

### PR DESCRIPTION
**Pull request by jeabakker**
ref #14075

### CODE REVIEW

**Comment by jeabakker** [comment]
Need #14130 first

File: engine/lib/input.php [hunk 1]
```diff
@@ -196,7 +194,7 @@ function elgg_is_empty($value) {
 function _elgg_htmlawed_tag_post_processor($element, $attributes = false) {
 	if ($attributes === false) {
 		// This is a closing tag. Prevent further processing to avoid inserting a duplicate tag
-		return "</${element}>";
+		return "</{$element}>";
```
Comment (jdalsem) [inline engine/lib/input.php:197]: !

Comment (jdalsem) [reply engine/lib/input.php:197]: should there be no tests?

File: engine/lib/input.php [hunk 2]
```diff
@@ -228,25 +226,25 @@ function _elgg_htmlawed_tag_post_processor($element, $attributes = false) {
...
 	}
 
 	// Some WYSIWYG editors do not like tags like <p > so only add a space if needed.
 	if ($string = trim($string)) {
-		$string = " $string";
+		$string = " {$string}";
 	}
 
-	$r = "<$element$string>";
+	$r = "<{$element}{$string}>";
```
Comment (jdalsem) [inline engine/lib/input.php:247]: can directly return

File: mod/pages/classes/ElggPage.php [hunk 3]
```diff
@@ -62,7 +62,7 @@ public function getTopParentEntity() {
 	 */
 	public function getParentEntity() {
 		
-		if (empty($this->parent_guid)) {
+		if (empty($this->parent_guid) || !is_int($this->parent_guid)) {
```
Comment (jdalsem) [inline mod/pages/classes/ElggPage.php:65]: why not cast to int?

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: engine/lib/input.php (hunk 1 above)
Comment (jeabakker) [reply engine/lib/input.php:197]: part of a callback of HTMLAwed not going to test this function specificly

**Review by jdalsem — APPROVED** [review]

**Pull request merged**